src/main/java/com/wolox/training/models/User.java

```diff
@@ -19,36 +22,53 @@
 @Table(name = "users")
 public class User {
 
-  public Long getId() {
-    return id;
+  @Id
+  @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "USER_SEQ")
+  @SequenceGenerator(name = "USER_SEQ", sequenceName = "USER_SEQ")
+  private Long id;
+
+  @Column(nullable = false, unique = true)
+  private String userName;
+
+  @Column(nullable = false)
+  private String name;
+
+  @Column(nullable = false)
+  private LocalDate birthDate;
+
+  @ManyToMany(cascade = {CascadeType.MERGE, CascadeType.REFRESH})
+  private List<Book> books;
+
+  public User() {
+    books = new ArrayList<>();
   }
 
-  public void setId(Long id) {
-    this.id = id;
+  public Long getId() {
+    return id;
   }
 
   public String getUserName() {
     return userName;
   }
 
   public void setUserName(String userName) {
-    this.userName = userName;
+    this.userName = checkNotNull(userName, Constant.NOT_NULL_MESSAGE, "userName");
   }
```

Comment:
Add check so that the attributes userName, name and birthDate are not empty.
Example:
Preconditions.checkArgument(!name.isEmpty(), "The name field cannot be empty");

Remember to add this validation after checking if it is null.